File: luojia/luojia-parent/easyBuy-portal-web/src/main/java/com/easybuy/portal/controller/UserController.java
package com.easybuy.portal.controller;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.alibaba.dubbo.config.annotation.Reference;
import com.easybuy.entity.Result;
import com.easybuy.pojo.TbUser;
import com.easybuy.service.UserService;

/**
 * controller
 * @author Romantic
 * @CreateDate 2020年4月11日
 * @Description
 */
@RestController
@RequestMapping("/user")
public class UserController {

	@Reference
	private UserService userService;
	
	
	/**
	 * 增加
	 * @param user
	 * @return
	 */
	@RequestMapping("/add")
	public Result add(@RequestBody TbUser user,String code){
		//判断验证码是否正确
		boolean checkSmsCode = userService.checkSmsCode(user.getPhone(), code);
		if(checkSmsCode==false){
			return new Result(false, "验证码输入错误！");
		}
		
		try {
			userService.add(user);
			return new Result(true, "增加成功");
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false, "增加失败");
		}
	}
	
	/**
	 * 修改
	 * @param user
	 * @return
	 */
	@RequestMapping("/update")
	public Result update(@RequestBody TbUser user){
		try {
			userService.update(user);
			return new Result(true, "修改成功");
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false, "修改失败");
		}
	}	
	
	/**
	 * 获取实体
	 * @param id
	 * @return
	 */
	@RequestMapping("/findOne")
	public TbUser findOne(Long id){
		return userService.findOne(id);		
	}
	
	
	@RequestMapping("/sendCode")
	public Result sendCode(String phone){
		
		try {
			userService.createSmsCode(phone);
			return new Result(true, "短信验证码发送成功");
		} catch (Exception e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
			return new Result(false, "短信验证码发送失败");
		}
	}
	
	
}

File: luojia/luojia-parent/easyBuy-portal-web/src/main/webapp/js/controller/userController.js
 //控制层 
app.controller('userController' ,function($scope,$controller   ,userService){	
	
	
	//查询实体 
	$scope.findOne=function(id){				
		userService.findOne(id).success(
			function(response){
				$scope.entity= response;					
			}
		);				
	}
	
	$scope.entity={};
	
	//注册
	$scope.reg=function(){		
		
		//判断两次输入的密码是否一致
		if($scope.entity.username=='' || $scope.entity.username==null){
			alert("请输入用户名");
			return ;
		}
		
		if($scope.entity.password=='' || $scope.entity.password==null){
			alert("请输入密码");
			return ;
		}
		
		if($scope.password=='' || $scope.password==null ){
			alert("请输入确认密码");
			return ;
		}
		
		if($scope.entity.password!=$scope.password  ){
			alert("两次输入的密码不一致!");
			return ;
		}	
		
		userService.add( $scope.entity ,$scope.smscode ).success(
			function(response){
				alert(response.message);
			}		
		);				
	}
	
	//发送短信验证码
	$scope.sendCode=function(){
		
		userService.sendCode( $scope.entity.phone  ).success(
			function(response){
				alert(response.message);				
			}			
		);		
	}
	
});	
